src/Microsoft.AspNet.Razor/Parser/HtmlMarkupParser.Block.cs

```diff
@@ -1136,5 +1136,24 @@ private void EndTagBlock(Stack<Tuple<HtmlSymbol, SourceLocation>> tags, bool com
             }
             Output(SpanKind.Markup);
         }
+
+        internal static bool IsValidAttributeNameSymbol(HtmlSymbol symbol)
+        {
+            if (symbol == null)
+            {
+                return false;
+            }
+
+            var symbolType = symbol.Type;
+            return symbolType != HtmlSymbolType.WhiteSpace &&
+                symbolType != HtmlSymbolType.NewLine &&
+                symbolType != HtmlSymbolType.CloseAngle &&
+                symbolType != HtmlSymbolType.OpenAngle &&
+                symbolType != HtmlSymbolType.ForwardSlash &&
+                symbolType != HtmlSymbolType.DoubleQuote &&
+                symbolType != HtmlSymbolType.SingleQuote &&
+                symbolType != HtmlSymbolType.Equals &&
+                symbolType != HtmlSymbolType.Unknown;
```

Comment:
`HtmlSymbolType` doesn't line up with HTML character classification. This expression allows `HtmlSymbolType.Transition`, `RazorComment`, `RazorCommentStar`, and `RazorCommentTransition`. None of those seem correct for attribute names in the Razor context.

Further `HtmlSymbolType.Text` (currently allowed) indicates _any_ non-whitespace non-symbol series of characters. Need to exclude control characters and unknown Unicode characters when that's the `symbolType`.